server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -209,6 +209,40 @@ public ImmutableList<Block> getInProgressBlocks() {
     return currentBlockList;
   }
 
+  @Override
+  public boolean shouldDisplayEligibilityMessage() {
+    return hasAnsweredEligibilityQuestions() && hasGatingEligibilityEnabledOrEligible();
+  }
+
+  /** Returns whether eligibility is gating or the application is eligible. */
+  private boolean hasGatingEligibilityEnabledOrEligible() {
+    if (programDefinition.eligibilityIsGating()) {
+      return true;
+    }
+    return !isApplicationNotEligible();
```

Comment:
That works! I thought since the `isApplicationEligible` function returns true either if the predicate evaluates to true or if there is no eligibility predicate, I thought there would be issues, but it seems this behaves the same way.